packages/flutter_test/lib/src/controller.dart

```diff
@@ -779,41 +810,142 @@ abstract class WidgetController {
+  /// console even if the test is logging output from the application.
+  @protected
+  void printToConsole(String message) {
+    debugPrint(message);
+  }
+
   // GEOMETRY
 
   /// Returns the point at the center of the given widget.
-  Offset getCenter(Finder finder) {
-    return _getElementPoint(finder, (Size size) => size.center(Offset.zero));
+  ///
+  /// If `warnIfMissed` is true (the default is false), then the
+  /// returned coordinate is checked to see if it would actually hit
+  /// the specified element, and if not, a warning is printed to the
+  /// console.
+  Offset getCenter(Finder finder, { bool warnIfMissed = false, String callee = 'getCenter' }) {
+    return _getElementPoint(finder, (Size size) => size.center(Offset.zero), warnIfMissed: warnIfMissed, callee: callee);
   }
 
   /// Returns the point at the top left of the given widget.
-  Offset getTopLeft(Finder finder) {
-    return _getElementPoint(finder, (Size size) => Offset.zero);
+  Offset getTopLeft(Finder finder, { bool warnIfMissed = false, String callee = 'getTopLeft' }) {
+    return _getElementPoint(finder, (Size size) => Offset.zero, warnIfMissed: warnIfMissed, callee: callee);
   }
 
   /// Returns the point at the top right of the given widget. This
   /// point is not inside the object's hit test area.
-  Offset getTopRight(Finder finder) {
-    return _getElementPoint(finder, (Size size) => size.topRight(Offset.zero));
+  Offset getTopRight(Finder finder, { bool warnIfMissed = false, String callee = 'getTopRight' }) {
+    return _getElementPoint(finder, (Size size) => size.topRight(Offset.zero), warnIfMissed: warnIfMissed, callee: callee);
   }
 
   /// Returns the point at the bottom left of the given widget. This
   /// point is not inside the object's hit test area.
-  Offset getBottomLeft(Finder finder) {
-    return _getElementPoint(finder, (Size size) => size.bottomLeft(Offset.zero));
+  Offset getBottomLeft(Finder finder, { bool warnIfMissed = false, String callee = 'getBottomLeft' }) {
+    return _getElementPoint(finder, (Size size) => size.bottomLeft(Offset.zero), warnIfMissed: warnIfMissed, callee: callee);
   }
 
   /// Returns the point at the bottom right of the given widget. This
   /// point is not inside the object's hit test area.
-  Offset getBottomRight(Finder finder) {
-    return _getElementPoint(finder, (Size size) => size.bottomRight(Offset.zero));
+  Offset getBottomRight(Finder finder, { bool warnIfMissed = false, String callee = 'getBottomRight' }) {
+    return _getElementPoint(finder, (Size size) => size.bottomRight(Offset.zero), warnIfMissed: warnIfMissed, callee: callee);
   }
 
-  Offset _getElementPoint(Finder finder, Offset sizeToPoint(Size size)) {
+  /// Whether warnings relating to hit tests not hitting their mark should be
+  /// fatal (cause the test to fail).
+  ///
+  /// Some methods, e.g. [tap], have an argument `warnIfMissed` which causes a
```

Comment:
I find the explanation in this paragraph of what warnIfMissed much cleared than what's documented on `tap` and friends. Unfortunately, most people will not read this here, but the paragraph on `tap` and friends. Maybe use this explanation in the other locations?

Reply:
i've tried to use text more similar to this text in the other locations